pymatviz/ptable.py

```diff
@@ -160,6 +162,231 @@ def count_elements(
     return srs
 
 
+def ptable_diags(
```

Comment:
maybe long-term we want to support more than 2 triangles per tile (i.e. 4)? in that case, maybe it makes sense to give this a more general name like `ptable_heatmap_split_tiles`? alternatively, maybe this shouldn't be a separate function but integrated into `ptable_heatmap()`? then the function would switch behavior based on the number of values per-element and raise ValueError if it's more than 4.

Reply:
Yes it makes sense to merge it into heatmap but I would need to think about how to switch in this case.

I thought about supporting arbitrary/dynamic number of splits (depending on the length of the list passed in for each element) actually, even allowing rotating the separation line (image a circle being split into several parts and then cropped as a rectangle)...But you know, could think of doesn't mean could implement, so I ended up limiting to only two. 

But it certainly would be good to add more options (4 for example), I would implement this tomorrow.

Meanwhile ptable plotters have heavily repeated code and I'm not sure if we could do something about it.

Reply:
> Meanwhile ptable plotters have heavily repeated code and I'm not sure if we could do something about it.

that's true. i've been putting off cleaning this up for a while. i think the simplest way forward would be to identify common chunks of code and extract into a helper function that is called by the different `ptable_heatmaps`.

Reply:
> i think the simplest way forward would be to identify common chunks of code and extract into a helper function that is called by the different `ptable_heatmaps`

I know it's easier said that done, maybe we should try to separate them into the periodic table itself and the nested plotter. I might have a go at this later.

Reply:
I just opened #133 for more detailed discussion on this :)